tree/treeplayer/inc/ROOT/TDFInterface.hxx

```diff
@@ -1235,8 +1252,8 @@ private:
          upcastNode, fImplWeakPtr, fValidCustomColumns, fDataSource);
       const auto thisTypeName = "ROOT::Experimental::TDF::TInterface<" + upcastInterface.GetNodeTypeName() + ">";
       return TDFInternal::JitTransformation(&upcastInterface, transformation, thisTypeName, nodeName, expression,
-                                            branches, customColumns, tmpBookedBranches, tree, returnTypeName,
-                                            fDataSource);
+                                            aliasMap, branches, customColumns, tmpBookedBranches, tree,
+                                            returnTypeName, fDataSource);
    }
```

Comment:
I don't think alias mapping belongs in `JitTransformation`. What do you think about factoring out a function `ApplyAliases` from `FindUsedColumnNames` and also calling it right before `CallJitTransformation`? So `JitTransformation` always receives real column and does not have to know about aliases at all.

Reply:
I realise the mapping here seems "too much". Ideally, TLoopManager should have an "AddColumnAlias" and a "GetAliasedColumn" method only. 
I think the issue here is that the expression may contain aliases. If we want to compile it, we need to deal with those while preparing the code to be jitted. A replace of alias->real column name in the expression will lead to a lot of trouble.

Reply:
ah you're right. then maybe we could just use a list of `validColumnNames` here (local list, see main review comment) and leave alias application to the jitted, templated call that `JitTransformation` produces? This is what we do for `Define`d columns

Reply:
Do you mean interrogate the alias mapping from within jitted code?

Reply:
agh, alright you need both old names and new names, the former to create the jitted string and the latter to get the column type. disregard my comments here :smile: